tests/ballerina-unit-test/src/test/resources/test-src/types/table/table_type.bal

```diff
@@ -1557,3 +1561,36 @@ function getXMLConversionResult(table|error tableOrError) returns xml {
     return retVal;
 }
 
+function testSelectQueryWithCursorTable() returns (int | error) {
+    h2:Client testDB = new({
+        path: "./target/tempdb/",
+        name: "TEST_DATA_TABLE_H2",
+        username: "SA",
+        password: "",
+        poolOptions: { maximumPoolSize: 1 }
+    });
+
+    table<IntData> t1 = check testDB->select("SELECT int_type from DataTable WHERE row_id = 1", IntData);
+    table<IntData> t1Copy = from t1 select *;
+    testDB.stop();
+    return 0;
+}
+
+function testJoinQueryWithCursorTable() returns (int | error) {
+    h2:Client testDB = new({
+        path: "./target/tempdb/",
+        name: "TEST_DATA_TABLE_H2",
+        username: "SA",
+        password: "",
+        poolOptions: { maximumPoolSize: 2 }
+    });
+
+    table<IntData> t1 = check testDB->select("SELECT int_type from DataTable WHERE row_id = 1", IntData);
+    table<IntData> t2 = check testDB->select("SELECT int_type from DataTable WHERE row_id = 1", IntData);
```

Comment:
Shall we remove "check" and use type checking here since theres a possibility of endpoint not being stopped in case an error is thrown while "check"ing.